### PR TITLE
Other picks

### PR DESCRIPTION
**Pull request by DanCesca**
Hello! With this pull request I have:

- added new samples/products to the database 
- modified the logic on Product Page 
- modified the routes to be able to use the sample id for the Other Picks on Product Page
- added Other Picks to Product Page
- finalized style of Product Page
- added a header component testing

Thank you very much in advance for your feedback and review!

### CODE REVIEW

File: client/src/pages/product.jsx [hunk 1]
```diff
@@ -120,3 +145,42 @@ const ErrorImage = styled.img`
 const ErrorMessage = styled.p`
   padding: 1rem;
 `;
+
+const OtherPicks = styled.ul`
```
Comment (PReichetanz) [inline client/src/pages/product.jsx]: The naming does not include an hint that this is a list.

File: client/src/pages/product.jsx [hunk 2]
```diff
@@ -120,3 +145,42 @@ const ErrorImage = styled.img`
...
+  align-items: center;
+  margin-bottom: 3rem;
+`;
+
+const OtherPicksTitle = styled.h2`
+  font-size: 1.3rem;
+  margin-bottom: 1rem;
+`;
+
+const OtherPicksContainer = styled.li`
```
Comment (PReichetanz) [inline client/src/pages/product.jsx]: The naming does not include an hint that this is a list item, I assumed a `div` or something similar.

File: client/src/components/TasteCard.jsx [hunk 3]
```diff
@@ -15,11 +15,14 @@ function TasteCard({ taste }) {
...
-          <SampleImage src={taste.sample.image} alt={taste.sample.name} />
+          <SampleImage src={taste.sample[0].image} alt={taste.sample[0].name} />
           <SampleText>
-            {taste.sample.name}: {taste.sample.flavors}
+            {taste.sample[0].name}: {taste.sample[0].flavors}
           </SampleText>
-          <StyledLink to={`/product/${taste.id}`}>
+          <StyledLink
+            onClick={() => window.scrollTo({ top: 0, behavior: 'smooth' })}
+            to={`/product/${taste.id}/${taste.sample[0].id}`}
```
Comment (PReichetanz) [inline client/src/components/TasteCard.jsx:24]: I don't understand this: Is it a ScrollToTop-Button or a Link to another page?

File: client/src/App.spec.jsx [hunk 4]
```diff
@@ -1,26 +1,27 @@
 import { render, screen } from '@testing-library/react';
 import TasteCard from './components/TasteCard';
 import App from './App';
+import {BrowserRouter as Router} from 'react-router-dom';
 
 describe('App', () => {
   it('renders three articles with an image, a title and a profile description', () => {
```
Comment (PReichetanz) [inline client/src/App.spec.jsx]: You don't search the screen for articles below and I didn't find an article in your code, so where are the three articles?

Besides, the `<App />` renders the Home page by default, so these tests should be either part of this page or (better, I think) part of the component within the Home page where these articles (or image, title and profile description - looks like they are part of the TasteCard.js?) are.

**Review by PReichetanz — COMMENTED** [review]
Looks good, I added some issues to consider 🚀

File: client/src/pages/product.jsx (hunk 1 above)
Comment (DanCesca) [reply client/src/pages/product.jsx]: Well, is that really so relevant? Anyways, I changed the name to OtherPicksList...

File: client/src/pages/product.jsx (hunk 2 above)
Comment (DanCesca) [reply client/src/pages/product.jsx]: Ok, changed it to OtherPicksItem

File: client/src/pages/product.jsx (hunk 1 above)
Comment (PReichetanz) [reply client/src/pages/product.jsx]: Let's say you now know what html element the styled component `OtherPicks` is, but believe me, when you have another look at your code in four weeks, you won't. This is why it is important to name everything in a proper manner.

File: client/src/components/TasteCard.jsx (hunk 3 above)
Comment (DanCesca) [reply client/src/components/TasteCard.jsx:24]: It is both. :)

File: client/src/App.spec.jsx (hunk 4 above)
Comment (DanCesca) [reply client/src/App.spec.jsx]: Exactly, the App renders the Homepage and the Homepage renders three cards. So I changed the description of the test to be 'renders three cards'. Thanks!

File: client/src/pages/product.jsx (hunk 1 above)
Comment (DanCesca) [reply client/src/pages/product.jsx]: I definitely get your point now! ;)

**Pull request merged**